### DISCUSSION TITLE
Including A Blog Within My Website

### DISCUSSION THREAD

**Discussion by samarzayan** [quarto-dev/quarto-cli#2949] [Q&A] [answered]
Hi there! This is a very noob-level question. My apologies in advance. Just began learning how to use Quarto.

I am creating a personal website and wish to create a blog within it. Should I do [this](https://quarto.org/docs/websites/website-blog.html) within the main website directory (mysite)?

Thanks in advance.

**Comment by dragonstyle** [comment, marked as answer]
Welcome! If you're adding a Quarto blog to an existing Quarto website, it should be largely about creating a 'listing' page. Just make a `blog.qmd` page with this yaml:

```markdown
---
title: "My Blog"
listing:
  contents: posts
  sort: "date desc"
  type: default
  categories: true
---
```

And the place your post `qmd` files in the posts folder (including in subfolders). Learn more here: https://quarto.org/docs/websites/website-listings.html

Does this make sense?

**Reply by samarzayan** [reply]
Yes, makes sense. Thanks so much.

**Comment by JosiahParry** [comment]
This is somewhat helpful. There are utilities in packages like `blogdown` and `hugodown` that will create a new template for you e.g. `blogdown::new_post()` which creates a new directory with an associated `index.md` file. Is this structure supported and automated with quarto?   Viewing the `quarto --help` there don't seem to be any utilities for this. 

<img width="793" alt="image" src="https://user-images.githubusercontent.com/13534508/201534801-77147f34-191e-48d4-a699-aa7f32a79b6b.png">

Does quarto support the structure of having 1 folder per post? Are there utilities to create a new post with the appropriate organization?

**Reply by smithjd** [reply]
Here's a nice solution to creating blog post structures automatically:

[Tom Mock blog](https://themockup.blog/posts/2022-11-08-use-r-to-generate-a-quarto-blogpost/)
